core/src/apps/monero/xmr/crypto/__init__.py

```diff
@@ -304,3 +304,21 @@ def xor8(buff, key):
     for i in range(8):
         buff[i] ^= key[i]
     return buff
+
+
+def cmp_arr(a, b):
```

Comment:
Do we really need this function? (Micro)Python can compare arrays and tuples just fine:

```
>>> [1,2,3] <= [1,2,3]
True
>>> b"123" <= b"123"
True
>>> [1,2,3] < [1,2,3]
False
>>> b"123" < b"123"
False
>>> [1,2,3] < [1,2,4]
True
>>> b"123" < b"124"
True
>>> None < [1,2,3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'NoneType' and 'list'
>>> None < b"123"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'NoneType' and 'bytes'
```

The only problem might be if you really require an exception in case the input lengths are different, because that just works fine with comparison operators:

```
>>> [1,2,3] < [1,2,3,4]
True
>>> [1,2,3,4] < [1,2,3]
False
>>> b"123" < b"1234"
True
>>> b"1234" < b"123"
False
```

In this case we might want to add an extra check just for the length.

Reply:
there is a problem with byte array

```python
>>> a=bytearray([1,2])
>>> b=bytearray([1,3])
>>> a<b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported types for __lt__: 'bytearray', 'bytearray'
```

Reply:
Ah, I was looking at `get_ki_from_vini` at thought it returns bytes, which is not the case. There is something off with the typing. Maybe we can rewrite the function like this and the comparison will start working?

```
def get_ki_from_vini(vini_bin: bytes) -> bytes:
    return bytes(vini_bin[-32:])
```

Reply:
I wanted to avoid memory fragmentation by redundant allocations. However, this is the only usage of the `cmp_arr` function, so implications by reallocating 32bytes to `bytes` should be OK I guess.

Reply:
But bytearray was maybe reallocating as well. OK I will do the change.

Reply:
Thanks!